File: 0020-valid-parentheses/0020-valid-parentheses.js
/**
 * @param {string} s
 * @return {boolean}
 */
const isValid = (str) => {
    const brackets = str.split(''),
          stack = [],
          pairs = {')':'(','}':'{',']':'['};
    for (let i = 0; i < brackets.length; i++) {
        const pair = pairs[brackets[i]];
        if (!pair) {
            stack.push(brackets[i])
        } else if (stack.pop() !== pair) {
            return false;
        }
    }
    return !stack.length;
};

File: 0020-valid-parentheses/0020-valid-parentheses.go
func isValid(s string) bool {
	pairs := map[rune]rune{']': '[', ')': '(', '}': '{'}
	stack := []rune{}
	for _, rn := range s {
		if pair, ok := pairs[rn]; ok {
			if len(stack) == 0 || stack[len(stack)-1] != pair {
				return false
			}
			stack = stack[:len(stack)-1]
		} else {
			stack = append(stack, rn)
		}
	}
	return len(stack) == 0
}